components/sidebar/sidebar_service.cc

```diff
@@ -545,7 +545,7 @@ std::vector<SidebarItem> SidebarService::GetDefaultSidebarItems() const {
   for (const auto& item_type : SidebarService::kDefaultBuiltInItemTypes) {
     if (auto item = GetBuiltInItemForType(item_type);
         item.built_in_item_type != SidebarItem::BuiltInItemType::kNone) {
-      items.push_back(item);
+      items.push_back(std::move(item));
```

Comment:
pls make sure `item` type has move constructors and move assignment

Reply:
I thought SidebarItem has them (created implicitel by compiler) as there is no build error.
Am I right?

Reply:
> I thought SidebarItem has them (created implicitely by compiler) as there was no build error. Am I right?

sadly, no. https://en.cppreference.com/w/cpp/language/move_constructor, see `Implicitly-declared move constructor`.

compiler does the rvalue cast to call `SidebarItem::SidebarItem(&&)`, doesn't find it, because a user-provided copy assignment is declared, fallbacks to calling `SidebarItem::SidebarItem(const &)`. no error here from the compiler perspective.

Reply:
`SidebarItem` has only default ctor and dtor. and I thought it doesn't have user declared ctor and dtor.
```
SidebarItem::SidebarItem() = default;
SidebarItem::~SidebarItem() = default;
```
According to the `Implicitly-declared move constructor`, implicit move ctor can be declared in this case.
Or above `default`ed one is considered as user declared?

Reply:
huh, you're right! I've mistaken `bool operator==(const SidebarItem& item) const;` with an assignment operator.
In this case we're good.

Reply:
Ok, default move/copy ctor/assign are added :)